app/core/chat/task_states/list_movies_state.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from typing import List
+
+from app.core.chat.dialogue_structs.intent import IntentEnum
+from app.core.chat.dialogue_structs.slot_mapping import SlotMapping
+from app.core.chat.nlg.nlg import NLG
+from app.core.chat.task_states.task_state import TaskState, NoSlotsToRequest, NoSlotsToSuggest
+
+
+@dataclass
+class ListMoviesState(TaskState):
+    def __init__(self):
+        super().__init__(
+            IntentEnum.LIST_MOVIES,
+            slots=[
+                SlotMapping(
+                    name="date",
+                    description="the date of the screening of the movie",
+                    _info_template="The screening happens on {}",
+                    _request_template="What is the date of the screening of the movie?",
+                    is_required=True
+                ),
+                SlotMapping(
+                    name="from_hour",
+                    description="beginning hour of requested screenings of the movie",
+                    _info_template="The screening are listed from {}",
+                    _request_template="From what hour do you want to list the screenings?",
+                    is_required=False
+                ),
+                SlotMapping(
+                    name="to_hour",
+                    description="ending hour of requested screenings of the movie",
+                    _info_template="The screening are listed until {}",
+                    _request_template="Until what hour do you want to list the screenings?",
+                    is_required=False
+                ),
+                SlotMapping(
+                    name="date",
+                    description="the date of the screening of the movie",
+                    _info_template="The screening happens on {}",
+                    _request_template="What is the date of the screening of the movie?",
+                    is_required=False
+                ),
+                SlotMapping(
+                    name="genre",
+                    description="genre of the movie",
+                    _info_template="The movie is a {}",
+                    _request_template="What genre of movies do you want to list?",
+                    is_required=False
+                ),
+            ]
+        )
+        self.suggestions_already_made = False
+
+    def generate_next_response(self, nlg: NLG) -> str:
+        try:
```

Comment:
Should this be a try-catch instead of a simple if-else? Just curious if there is any advantage, cause `except NoSlotsToRequest: pass` seems like overkill.